test/factree/pathfinder_test.rb

```diff
@@ -0,0 +1,60 @@
+require 'test_helper'
+
+describe Factree::Pathfinder do
+  describe "#find_node_sequence" do
+    let(:facts) do
+      {
+        has_unibrow?: true,
+        lives_in_trash_can?: false
+      }
+    end
+
+    def root(facts)
+      facts.require :lives_in_trash_can?
+
+      if facts[:lives_in_trash_can?]
+        Factree::Conclusion.new(:oscar)
+      else
+        unibrow_decision(facts)
+      end
+    end
+
+    def unibrow_decision(facts)
+      if facts[:has_unibrow?]
+        Factree::Conclusion.new(:bert)
+      else
+        Factree::Conclusion.new(:ernie)
+      end
+    end
+
+    let(:root_proc) { method :root }
+    subject { Factree::Pathfinder.find facts, &root_proc }
+
+    it "knows all of the facts that were required along the way" do
+      subject.required_facts.must_equal [:lives_in_trash_can?, :has_unibrow?]
+    end
+
+    it "keeps going until it reaches a conclusion" do
+      subject.complete?.must_equal true
```

Comment:
I went through and changed the tests to use [assertions](http://docs.seattlerb.org/minitest/Minitest/Assertions.html).